test/spec/spectest.cpp

```diff
@@ -128,85 +160,53 @@ parseValueList(const rapidjson::Value &Args) {
-          I64x2[I] = std::stoull(ValueNode[I].Get<std::string>());
-        }
-      } else if (LaneType == "i32"sv || LaneType == "f32"sv) {
-        using uint32x4_t = uint32_t __attribute__((vector_size(16)));
-        uint32x4_t I32x4 = {0};
-        for (rapidjson::SizeType I = 0; I < 4; ++I) {
-          I32x4[I] = std::stoul(ValueNode[I].Get<std::string>());
-        }
-        I64x2 = reinterpret_cast<WasmEdge::uint64x2_t>(I32x4);
-      } else if (LaneType == "i16"sv) {
-        using uint16x8_t = uint16_t __attribute__((vector_size(16)));
-        uint16x8_t I16x8 = {0};
-        for (rapidjson::SizeType I = 0; I < 8; ++I) {
-          I16x8[I] = static_cast<uint16_t>(
-              std::stoul(ValueNode[I].Get<std::string>()));
-        }
-        I64x2 = reinterpret_cast<WasmEdge::uint64x2_t>(I16x8);
-      } else if (LaneType == "i8"sv) {
-        using uint8x16_t = uint8_t __attribute__((vector_size(16)));
-        uint8x16_t I8x16 = {0};
-        for (rapidjson::SizeType I = 0; I < 16; ++I) {
-          I8x16[I] =
-              static_cast<uint8_t>(std::stoul(ValueNode[I].Get<std::string>()));
-        }
-        I64x2 = reinterpret_cast<WasmEdge::uint64x2_t>(I8x16);
-      }
-      Result.emplace_back(I64x2);
-      ResultTypes.emplace_back(WasmEdge::ValType::V128);
-    } else {
-      assumingUnreachable();
     }
   }
   return {Result, ResultTypes};
 }
 
 // Helper function to parse parameters from json to vector of string pair.
 std::vector<std::pair<std::string, std::string>>
-parseExpectedList(const rapidjson::Value &Args) {
+parseExpectedList(const simdjson::dom::array &Args) {
   std::vector<std::pair<std::string, std::string>> Result;
-  Result.reserve(Args.Size());
-  for (const auto &Element : Args.GetArray()) {
-    const auto &Type = Element["type"].Get<std::string>();
-    const auto &ValueNode = Element["value"];
-    if (ValueNode.IsString()) {
-      Result.emplace_back(Type, ValueNode.Get<std::string>());
-    } else if (ValueNode.IsArray()) {
-      const auto LaneType = Element["lane_type"].Get<std::string>();
-      std::string Value;
-      for (auto Iter = ValueNode.Begin(); Iter != ValueNode.End(); ++Iter) {
-        Value += Iter->Get<std::string>();
-        Value += ' ';
+  Result.reserve(Args.size());
+  simdjson::dom::array ValueNodeArray;
+  for (simdjson::dom::object Element : Args) {
+    std::string_view Type = Element["type"];
+    if (!Element["value"].get(ValueNodeArray)) {
+      std::string StrValue;
+      std::string_view LaneType = Element["lane_type"];
+      for (std::string_view X : ValueNodeArray) {
+        StrValue += std::string(X);
+        StrValue += ' ';
       }
-      Value.pop_back();
-      Result.emplace_back(Type + LaneType, std::move(Value));
+      StrValue.pop_back();
+      Result.emplace_back(std::string(Type) + std::string(LaneType), StrValue);
     } else {
-      assumingUnreachable();
+      std::string_view Value = Element["value"];
```

Comment:
The condition could be `array`, `string`, or `unexpected error`. This modification only maintains the first two conditions.

Reply:
Ohh, missed it. Will add another condition for the same.